bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/internal/BRC1HUartProcessor.java

```diff
@@ -78,21 +86,27 @@ private boolean isMessageComplete() {
     }
 
     public void chunkReceived(byte[] byteValue) {
-        this.uartMessages.add(byteValue);
-        if (isMessageComplete()) {
-
-            // Beyond this point, full message received
-            ByteArrayOutputStream bos = new ByteArrayOutputStream();
-
-            for (byte[] msg : uartMessages) {
-                if (msg.length > 1) {
-                    bos.write(msg, 1, msg.length - 1);
+        stateLock.lock();
+        try {
+            this.uartMessages.add(byteValue);
+            if (isMessageComplete()) {
+                logger.debug("Complete message received!");
+
+                // Beyond this point, full message received
+                ByteArrayOutputStream bos = new ByteArrayOutputStream();
+
+                for (byte[] msg : uartMessages) {
+                    if (msg.length > 1) {
+                        bos.write(msg, 1, msg.length - 1);
+                    }
                 }
-            }
 
-            this.uartMessages.clear();
+                this.uartMessages.clear();
 
-            this.responseListener.receivedResponse(bos.toByteArray());
+                this.responseListener.receivedResponse(bos.toByteArray());
```

Comment:
I think that calling this inside of a lock might risk a deadlock in the core. Can you call this outside of your lock/unlock block?

Reply:
I dont see other option. The goal here is to avoid the Byte Array Output Stream to be modified. Otherwise, I should maybe clone the BOS and call the received response after ?

Reply:
Copying the array would be preferred, yes.